types/ember/index.d.ts

```diff
@@ -3197,26 +3204,24 @@ declare module 'ember' {
          * the function will be invoked. If the property is not defined but the
          * object implements the `unknownProperty` method then that will be invoked.
          */
-        function get<T, K extends keyof T>(obj: ComputedPropertyGetters<T>, key: K): T[K];
-        function get<T, K extends keyof T>(obj: T, key: K): T[K]; // for dynamic K
+        function get<T, K extends keyof T>(obj: T, key: K): UnwrapComputedPropertyGetter<T[K]>;
         /**
          * Retrieves the value of a property from an Object, or a default value in the
          * case that the property returns `undefined`.
          */
         function getWithDefault<T, K extends keyof T>(
-            obj: ComputedPropertyGetters<T>,
+            obj: T,
             key: K,
-            defaultValue: T[K]
-        ): T[K];
-        function getWithDefault<T, K extends keyof T>(obj: T, key: K, defaultValue: T[K]): T[K]; // for dynamic K
+            defaultValue: UnwrapComputedPropertyGetter<T[K]>
+        ): UnwrapComputedPropertyGetter<T[K]>;
         /**
          * Sets the value of a property on an object, respecting computed properties
          * and notifying observers and other listeners of the change. If the
          * property is not defined but the object implements the `setUnknownProperty`
          * method then that will be invoked as well.
          */
         function set<T, K extends keyof T, V extends T[K]>(
-            obj: ComputedPropertySetters<T>,
+            obj: UnwrapComputedPropertySetters<T>,
```

Comment:
I think this is another inversion.

```ts
function set<T, K extends keyof T, V extends UnwrapComputedPropertySetter<T[K]>>(
  obj: t,
  key: K,
  value: V
): V;
```

Reply:
Can you guys contribute some test cases that validate the correct behavior?

Reply:
@dfreeman I don't believe that this is correct. Current tests fail w/ your suggested implementation, and pass with the current code

Reply:
It seems to work either way 🤷‍♂️ 

Would it be more correct for `set` to return the _get_ type?
```ts
function set<T, K extends keyof T>(
    obj: T,
    key: K,
    value: UnwrapComputedPropertySetter<T[K]>
): UnwrapComputedPropertyGetter<T[K]>;
```

Reply:
^ this works